File: Arc3/Core/Schema/Insight.cs
using System.Collections;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace Arc3.Core.Schema;

public class Insight {

  [BsonId]
  [BsonRepresentation(BsonType.String)]
  public string Id { get; set; }

  [BsonElement("type")]
  public string Type { get; set; }

  [BsonElement("date")]
  public long Date { get; set; }

  [BsonElement("tagline")]
  public string Tagline { get; set; }

  [BsonElement("guild_id")]
  public string GuildID { get; set; }

  [BsonElement("data")]
  public BsonDocument  Data { get; set; }

  [BsonElement("url")]
  public string Url { get; set; }

}

File: arc3-api/src/v1/db/Insight.js
const mongoose = require('mongoose');

const insightSchema = new mongoose.Schema({
  _id: String,
  type: String,
  date: String,
  tagline: String,
  guild_id: String,
  data: Object,
  url: String
}, {
  collection: "Insights"
})

const Insight = mongoose.model("Insight", insightSchema);

module.exports = Insight;